### PR TITLE
outsourcing 

### PR DESCRIPTION
**Pull request by gickowtf**
hey @Mrredstone5230 and @ireneybean i was looking for a way to outsource at least the special_pages like PV and Fuel. i think this is a great way. Especially since I actually have a lot of special pages on the todo list and so we can avoid a lot of "IFs".

### CODE REVIEW

**Comment by Mrredstone5230** [comment]
I mean, that's a good idea... But I think Irene was sort of already working on it (take a look at ad9355f, very interesting stuff!)

**Comment by ireneybean** [comment]
I am, but this is still a good idea. Don't wait on me :-) i think this will fit in fine with my approach.

File: custom_components/divoom_pixoo/pages/fuel.py [hunk 1]
```diff
@@ -16,12 +16,12 @@ def fuel(pixoo, hass, page_data: dict, FONT_PICO_8, FONT_GICKO, FIVE_PIX, ELEVEN
...
     darkgrey = (36, 36, 36)  #datetime bg
-    black = (0, 0, 0) #default title
-    white = (255, 255, 255)  #date
-    yellow = (255, 230, 0) #default bg + time
+    black = "(0, 0, 0)" #default title
+    white = "(255, 255, 255)"  #date
+    yellow = "(255, 230, 0)" #default bg + time
 
     for key in page_data.keys():  # Convert all values to strings. Avoids problems.
-        page_data[key] = str(page_data[key])
```
Comment (Mrredstone5230) [inline custom_components/divoom_pixoo/pages/fuel.py:24]: by removing this, it makes the for loop useless. You could probably remove it then..

File: custom_components/divoom_pixoo/pages/fuel.py [hunk 2]
```diff
@@ -36,15 +36,15 @@ def fuel(pixoo, hass, page_data: dict, FONT_PICO_8, FONT_GICKO, FIVE_PIX, ELEVEN
...
-    font_color = tuple(page_data.get('font_color', white))
-    bg_color = tuple(page_data.get('bg_color', yellow))
-    price_color = tuple(page_data.get('price_color', white))
-    title_color = tuple(page_data.get('title_color', black))
-    stripe_color = tuple(page_data.get('stripe_color', font_color))
+    font_color = eval(page_data.get('font_color', white))
+    bg_color = eval(page_data.get('bg_color', yellow))
+    price_color = eval(page_data.get('price_color', white))
+    title_color = eval(page_data.get('title_color', black))
+    stripe_color = eval(page_data.get('stripe_color', white))
```
Comment (Mrredstone5230) [inline custom_components/divoom_pixoo/pages/fuel.py:43]: isn't eval pretty unsafe? I mean, doing it like this trusts unfiltered user input... I totally get what you're doing, but I'm just a liiiiiitle bit scared with the security of doing this. Maybe we could move the color rendering of the components page to a different method? (Line 141 of the sensor.py I think?)

File: custom_components/divoom_pixoo/pages/fuel.py (hunk 1 above)
Comment (gickowtf) [reply custom_components/divoom_pixoo/pages/fuel.py:24]: I don't know why, but I am firmly convinced that I have reinstalled the str() -.-

File: custom_components/divoom_pixoo/pages/fuel.py (hunk 2 above)
Comment (gickowtf) [reply custom_components/divoom_pixoo/pages/fuel.py:43]: well in the end it's just a user input but you're right the eval() was a stopgap to get the tuple back. i think finding the solution in the sensor.py would be better. unfortunately it's already 11pm for me -.-

Comment (Mrredstone5230) [reply custom_components/divoom_pixoo/pages/fuel.py:43]: Alright, I can do this if you want. Go sleep!

Comment (gickowtf) [reply custom_components/divoom_pixoo/pages/fuel.py:43]: all right, many thanks for the quick response

Comment (gickowtf) [reply custom_components/divoom_pixoo/pages/fuel.py:43]:
```python 
import ast
value = "(255, 255, 255)"

result = ast.literal_eval(value)

if isinstance(result, tuple):
    pass
```

Maybe this is the solution

Comment (Mrredstone5230) [reply custom_components/divoom_pixoo/pages/fuel.py:43]: Well this annoys me, as it doesn't allow users to use the "new" color strings (with just writing the color's name).

Comment (Mrredstone5230) [reply custom_components/divoom_pixoo/pages/fuel.py:43]: Alright, you can check #34.

**Pull request merged**